core/store/src/trie/shard_tries.rs

```diff
@@ -16,7 +16,8 @@ pub struct ShardTries {
 }
 
 impl ShardTries {
-    pub fn new(storage: Arc<Store>, num_shards: ShardId) -> Self {
+    pub fn new(storage: Arc<Store>, num_shards: u64) -> Self {
+        debug_assert_ne!(num_shards, 0);
```

Comment:
shouldn't this be an actual assert? It cannot possibly work with 0 shards.

Reply:
++